mlflow/utils/search_utils.py

```diff
@@ -987,3 +916,105 @@ def _apply_reversor(experiment, key, ascending):
+
+    @classmethod
+    def _get_identifier(cls, identifier):
+        tokens = identifier.split(".", maxsplit=1)
+        if len(tokens) == 1:
+            key = tokens[0]
+            identifier = cls._ATTRIBUTE_IDENTIFIER
+        else:
+            entity_type, key = tokens
+            valid_entity_types = ("attribute", "tag", "tags")
+            if entity_type not in valid_entity_types:
+                raise MlflowException.invalid_parameter_value(
+                    f"Invalid entity type '{entity_type}'. "
+                    f"Valid entity types are {valid_entity_types}"
+                )
+            identifier = cls._TAG_IDENTIFIER if entity_type in ("tag", "tags") \
+                else cls._ATTRIBUTE_IDENTIFIER
+
+        key = cls._trim_backticks(cls._strip_quotes(key))
+        return {"type": identifier, "key": key}
+
+    @classmethod
+    def _get_comparison(cls, comparison):
+        stripped_comparison = [token for token in comparison.tokens if not token.is_whitespace]
+        cls._validate_comparison(stripped_comparison)
+        left, comparator, right = stripped_comparison
+        comp = cls._get_identifier(left.value)
+        comp["comparator"] = comparator.value.upper()
+        comp["value"] = cls._get_value(comp.get("type"), comp.get("key"), right)
+        return comp
+
+    @classmethod
+    def _get_value(cls, identifier_type, key, token):
+        if identifier_type == cls._TAG_IDENTIFIER:
+            if token.ttype in cls.STRING_VALUE_TYPES or isinstance(token, Identifier):
+                return cls._strip_quotes(token.value, expect_quoted_value=True)
+            raise MlflowException(
+                "Expected a quoted string value for "
+                "{identifier_type} (e.g. 'my-value'). Got value "
+                "{value}".format(identifier_type=identifier_type, value=token.value),
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        elif identifier_type == cls._ATTRIBUTE_IDENTIFIER:
+            if token.ttype in cls.STRING_VALUE_TYPES or isinstance(token, Identifier):
+                return cls._strip_quotes(token.value, expect_quoted_value=True)
+            elif isinstance(token, Parenthesis):
+                cls._check_valid_identifier_list(token)
+                if key != "run_id":
+                    raise MlflowException(
+                        "Only run_id attribute support compare with a list of quoted string values.",
+                        error_code=INVALID_PARAMETER_VALUE,
+                    )
+                run_id_list = cls._parse_list_from_sql_token(token)
+                # Because MYSQL IN clause is case in-sensitive, but all run_ids only contains lower case
+                # letters, so that we filter out run_ids containing upper case letters here.
+                run_id_list = [
+                    run_id for run_id in run_id_list if run_id.lower() == run_id
+                ]
```

Comment:
Do we really need this? Do users really use a query like `run_id IN ('AbC', 'EfG')?`

Reply:
I think this is more of a correctness issue.
e.g.
suppose run_id "135eeedb755f44e0a2ddde9ea999f573"
user type it as  `run_id IN ("135EEEdb755f44e0a2ddde9ea999f573", ...)` by accident, 
then the filter should be regarded as false.

@dbczumar What do you think ? If no need to handle this we can remove it.

Reply:
I think we should keep this here.